src/lib.rs

```diff
@@ -161,6 +161,15 @@ impl HttpMetricsLayer {
             .with_state(self.state.clone())
     }
 
+    /// Create a Router that has a customized listen metrics path
```

Comment:
that will works.

but I think it should implemented in the builder. 

and we just need to add a metrics_path field to the state.

then, the router builder can get it